### PR TITLE
chore: fix length assertion

### PR DESCRIPTION
**Pull request by peternhale**
### What does this PR do?
Loosens the assert when checking the number of record from an exact match to an "above" match. This assertion fails at times when the async bulk delete reports as complete, but is really not. This is a server side issue, so accounting for the possibility that this may crop up in a test.
### What issues does this PR fix or reference?
@W-0@

### CODE REVIEW

File: packages/plugin-data/package.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
     "test": "sf-test",
     "test:command-reference": "./bin/run commandreference:generate --erroronwarnings",
     "test:deprecation-policy": "./bin/run snapshot:compare",
-    "test:nuts": "nyc mocha \"**/*.nut.ts\" --slow 4500 --timeout 600000 --parallel",
+    "test:nuts": "nyc mocha \"**/*.nut.ts\" --slow 4500 --timeout 600000",
```
Comment (mshanemc) [inline packages/plugin-data/package.json:72]: why?

**Review by mshanemc — COMMENTED** [review]
why disable parallel?

**Comment by RodEsp** [comment]
I second Shane's question ^.

**Comment by peternhale** [comment]
> why disable parallel?
I'll revert

**Review by shetzel — APPROVED** [review]

**Review by RodEsp — APPROVED** [review]

**Pull request merged**